application-forum-ui/src/main/resources/ForumCode/Macros.xml

```diff
@@ -479,10 +479,24 @@
   #end
 #end
 ##
+#macro(checkIfHostNameIsValid $stringURL $result)
+  ## @since 12.8RC1
+  #set ($hostName = $urltool.getURLFromString($stringURL).getHost())
+  #if (!$hostName)
+    #set ($hostName = $stringURL.split('//')[1].split('/')[0].split(':')[0])
+  #end
+  #if ($hostName.contains('.'))
+    #set ($result = true)
+  #else
+    #set ($result = false)
+  #end
```

Comment:
We usually use `#setVariable`, check it [here](https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/macros.vm#L1202) (examples of usage in the same file).